### DISCUSSION TITLE
Thanks

### DISCUSSION THREAD

**Discussion by TekyTeky** [Novakasa/brickrail#158] [General]
I had a chance today to use the new master and just wanted to say thanks for all the updates. 
Having the alternative colour for the train makes a big difference on smaller screens… although I couldn’t find the setting at first.
The hover over the switches is good too and really like showing the linking of the reverse marker to the block. And clickability of the block markers. 
There was an issue when the layout hubs were updated whereby they didn’t hold their invert direction setting which confused me at first.
Was there anything else I should have noticed?

Plus I know you did a lot of other back end stuff too.. so thanks.
I know my “nice to have” list is pretty empty now, so what’s next. 😎

**Comment by Novakasa** [comment]
Thanks for the nice words!
Oh yeah I forgot about the switch invert setting, I did change some internal stuff about it, so it is now stored differently and so each invert setting was reset.

Regarding what's next... The past few days I fixed a lot of bugs, in the hope that I can do another release soon. Currently thinking about calling it a beta release. I unfortunately still know about a few issues but fixing them is sometimes really hard with the state of the code.

I have been using the Rust language in some other projects and tbh I am currently experimenting with how hard it would be to rewrite brickrail in there. Maintaining such a complex application in Godot and GDscript is honestly quite hard, it would really help a lot to have a robust codebase in Rust. It might also enable me to make it a pure web-app with web-bluetooth support and a Mac release might be easier. Of course that would be another project that can take years to reach the same maturity as this now, so I at least want to leave the current brickrail version in a quite usable state. I am also considering still adding the train scheduling feature, and of course if some bigger issues crop up I still can try to fix them.

**Reply by TekyTeky** [reply]
Mmm.. bugs.
As an IT application support tech they are the bane of my work life...but I neither cause or fix them.. lol

One thing that still troubles me is the reversing issue after the system has stopped. Yesterday a train was reversing to a block and I had placed it incorrectly on the track so I an forced an emergency stop. I just couldn't get it to go forward again, so had to turn it off.

It hasn't happened lately as I have mainly been testing out the new section and definitely not using Random mode.

Upgrading to Rust would be good long term I guess.. gives you better flexibility and options. 
Bluetooth on web pages sounds tricky.

Do it while you can, life is far too short to have regrets... that's my philosophy tip of the day... lol
